### PR TITLE
feat/1495 rest_client: renames JSONResponsePaginator to JSONLinkPaginator

### PR DESCRIPTION
**Pull request by willi-mueller**
### Description

Implements: https://github.com/dlt-hub/dlt/issues/1495

### Related Issues
- once this is merged and released, also merge the depending PR in verified-sources: https://github.com/dlt-hub/verified-sources/pull/523

### CODE REVIEW

**Review by rudolfix — CHANGES_REQUESTED** [review]
Looks good but this is a breaking change so what we need is
- JSONResponsePaginator as an alias or derive it directly from `JSONLinkPaginator`
- deprecate it, I think you can deprecate classes

what are you planning to do with `rest_api`? will it continue to work with old `dlt` or you will bump the required version?

**Comment by burnash** [comment]
@willi-mueller Looks good, I agree with @rudolfix we need to deprecate JSONResponsePaginator,

@rudolfix will something like this work?

```python
import warnings

class JSONLinkPaginator(BaseNextUrlPaginator):
    ...


class JSONResponsePaginator(JSONLinkPaginator):
    def __init__(self):
        warnings.warn("JSONResponsePaginator is deprecated and will be removed in version 1.0.0. Use JSONLinkPaginator instead.", DeprecationWarning, stacklevel=2)
        super().__init__()
```

**Review by burnash — CHANGES_REQUESTED** [review]
Great work, let's also deprecate the old class. See the comment above.

**Comment by willi-mueller** [comment]
> Great work, let's also deprecate the old class. See the comment above.

Thanks! I marked the old class as deprecated and rebased onto devel.

File: docs/website/docs/general-usage/http/overview.md [hunk 1]
```diff
@@ -58,11 +58,11 @@ Note that we do not explicitly specify the pagination parameters in the example.
...
 import dlt
 from dlt.sources.helpers.rest_client import RESTClient
-from dlt.sources.helpers.rest_client.paginators import JSONResponsePaginator
+from dlt.sources.helpers.rest_client.paginators import JSONLinkPaginator
 
 github_client = RESTClient(
     base_url="https://pokeapi.co/api/v2",
-    paginator=JSONResponsePaginator(next_url_path="next"),   # (1)
+    paginator=JSONLinkPaginator(next_url_path="next"),   # (1)
     data_selector="results",                                 # (2)
```
Comment (burnash) [inline docs/website/docs/general-usage/http/overview.md:66]:
```suggestion
    data_selector="results",                             # (2)
```

**Review by burnash — APPROVED** [review]
Looks good! Thank you @willi-mueller

**Pull request merged**